perf/config.py

```diff
@@ -1,33 +1,145 @@
-from typing import List
+import logging
+import sys
+import time
 from pathlib import Path
+from typing import List
+from typing import Optional
+from typing import Type
+from urllib.parse import urlparse
+
 import attr
+from constants import SEMGREP_URL
+from constants import SEMGREP_USER_AGENT
 from ruamel.yaml import YAML
-import time
 
-import logging
-import sys
 
 logger = logging.getLogger(__file__)
 logger.setLevel(logging.DEBUG)
 handler = logging.StreamHandler(stream=sys.stderr)
-handler.setFormatter(logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s'))
+handler.setFormatter(
+    logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+)
 logger.addHandler(handler)
 
-yaml = YAML(typ='rt')
+yaml = YAML(typ="rt")
+
 
 @attr.s
 class Repository(object):
     url: str = attr.ib(default="")
     commit_hash: str = attr.ib(default="HEAD")
 
+
+@attr.s
+class RuleConfig(object):
+    config_str: str = attr.ib(default="")
+
+    def _fetch_rule_config_from_url(self, rule_config_url: str) -> Optional[str]:
+        import requests
+
+        logger.info(f"Fetching config from '{rule_config_url}'")
+        try:
+            r = requests.get(
+                rule_config_url,
+                headers={"User-Agent": SEMGREP_USER_AGENT},
+                timeout=60,
+            )
+            r.raise_for_status()
+        except requests.Timeout:
+            logger.warning(
+                f"There was a timeout error trying to download the rule config at '{rule_config_url}'"
+            )
+            return None
+        except requests.HTTPError as e:
+            logger.warning(
+                f"There was an issue connecting to '{rule_config_url}'; error: {e}"
+            )
+            return None
+        return r.text
+
+    def _fetch_rule_config_from_path(self, path: Path) -> Optional[str]:
+        pass
```

Comment:
what's the main purpose of this function? 😲

Reply:
Oops, should've deleted this! I'll open another PR to remove.